docs/plugins.md

```diff
@@ -1,5 +1,202 @@
 # Plugin System
 
-The plugin system will allow to manipulate requests and responses inside a `HttpClient`.
+The plugin system allow to manipulate requests and responses inside an `HttpClient`.
```

Comment:
though "manipulate" is a bit misleading, as they are immutable. should we say "to look at requests and responses and replace them if needed"?